### DISCUSSION TITLE
Fluentd configuration

### DISCUSSION THREAD

**Discussion by subbeh** [ricsanfre/pi-cluster#132] [Q&A] [answered]
Hi,

Thanks so much for all the information first of all, I've been learning a lot trying to get this all to work in my own cluster. 
That said, I'm a bit stuck with the complexity of the logging stack. 

I've deployed elasticsearch, kibana, fluentd and fluentbit. I can see fluentd indexes in kibana, but there doesn't seem to be any data.

![image](https://github.com/ricsanfre/pi-cluster/assets/1278086/94f4cfe7-2bdc-43fa-bcfd-8b6218704ea2)

Looking at the fluentd logs, I'm getting these errors:

```
[warn]: #0 [out_es] Could not communicate to Elasticsearch, resetting connection and trying again. [400] {"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:12] [policy] unknown field [phases]"}],"type":"x_content_parse_exception","reason":"[1:12] [policy] unknown
 field [phases]"},"status":400}
```

Not really sure where the problem is. [This is my config](https://github.com/Subbeh/homelab/blob/main/k8s/apps/monitoring/fluentd/values.yaml). Any help or push in the right direction would be highly appreciated.

Thanks

**Comment by ricsanfre** [comment]
Hi @Subbeh, thanks for your feedback.

Regarding your issue. I cannot access to your config file (link seems to be broken), but just based on the information from the screenshot you attached from Kibana's DataView configuration, it seems that fluentd is being able to create indexes in ES, and so it should mean data. Did you manage to create the DataView, setting "fluentd-*" as "Index Pattern"? 

Regards
Ricardo

**Reply by subbeh** [reply]
Hi @ricsanfre, thanks for getting back to me.

I might be missing something obvious, but I don't think any data is actually loaded. This is my data view:

![image](https://github.com/ricsanfre/pi-cluster/assets/1278086/979c91e4-0f9d-4597-8ed7-3ea0c7b3cdc1)

And this is my data discovery view:

![image](https://github.com/ricsanfre/pi-cluster/assets/1278086/e544e2b3-d5bd-401c-9960-8a66ac10d099)

Apologies for my broken link, I did change quite a bit to my repository since I posted this. [This is the correct link](https://github.com/Subbeh/homelab/blob/main/k8s/apps/logging/fluentd/values.yaml) (don't waste your time, I did make quite a lot of customizations to make it work in my environment)

I might have to go back and start from scratch, but it's strange that the connection seems to work while no data is being loaded. That error message I posted is peculiar though.

**Reply by ricsanfre** [reply, marked as answer]
Hi @Subbeh, if you are using ES ILM policies you should upgrade ricsanfre/fluentd-aggregator docker image. Tag to use should be v1.15.3-debian-1.3. That version is the one supporting the creation of ILM policies. I had to upgrade it because of an error I found with required packages versions. See  [this comment in issue #107](https://github.com/ricsanfre/pi-cluster/issues/107#issuecomment-1464902646) for more details. The message error you posted is the one i was having.

This is documented here https://picluster.ricsanfre.com/docs/logging-forwarder-aggregator/#enabling-elasticsearch-ilm

Let me know if this change solves your issue.

Regards
Ricardo

**Reply by subbeh** [reply]
Amazing, thank you @ricsanfre! It's all working now. 

I'll be sure to pay more attention to the issues and documentation going forward..
